### PR TITLE
Adding the ability to paint on floors with brushes, and also the ability to use multiple crayons from a box while floorpainting

### LINKED ISSUE #35300: [Feature Request] Special UI for painting/graffiti if using a rainbow crayon.
When making paintings or graffiti with a crayon, it becomes a hassle really quick to switch colours, cause you have to save the painting, inhand the crayon, select 2 colours you haven't selected yet, click the painting menu again, paint and repeat until you have completed the painting. Requesting that you get access to a hex value/RGB + colour wheel while using the crayon.
[featurerequest]

### PR DESCRIPTION
**Pull request by Volttekka**
closes #35300
Credit to Anonymous user No.453861032 from the thread for the coding
The purpose of this pull request is the ability to paint on floors with a brush, as well as the ability to use all the crayons in a box of crayons to draw on the floor in the advanced graffiti.

Why it's good: this is a logical extension of the painting and floor graffiti systems, so there shouldn't be any issues.

This is my first time doing any sort of pull request, so lemme know if I goofed a bit on this
Edit: added changelog

 :cl:
  - rscadd: Thanks to >>453861032, you can now paint on floors with a paintbrush.

### CODE REVIEW

**Comment by kilozombie** [comment]
Did you test it ingame to see if it works?

**Comment by DeityLink** [comment]
what kilo said. other than that this is a good idea. (and thankfully for me compatible with the big paint PR I'm currently cooking)

**Comment by Eneocho** [comment]
Closes #35300.

**Comment by Eneocho** [comment]
Requesting an RGB/HEX input or something if you're painting with the rainbow crayon (as it can paint in all colours), or being able to add colours to an easel via crayon.

**Comment by SonixApache** [comment]
add `closes #35300` somewhere to your OP so the collabs can auto-close it on merge.
Add a changelog to the end of your OP, something like this
```
 :cl:
  - rscadd: Thanks to >>453861032, you can now paint on floors with a paintbrush.
```

**Comment by jellyveggie2** [comment]
Code seems good, and other than making the pull request straight from Bleeding-Edge rather than a separate branch, the PR itself seems good too. Not that that should be an issue in this case I think.

@kilozombie 
> Did you test it ingame to see if it works?

Looking at the thread, I'm not sure he's ever set up a local server before. I'll test it a bit and report with results.

**Comment by jellyveggie2** [comment]
Brush graffitti is working fine, but box-crayoning is adding the same crayon multiple times. Probably an easy fix.

File: code/modules/html_interface/paintTool/custom_painting_datum.dm [hunk 1]
```diff
@@ -31,6 +31,15 @@
 
 	// Painting with a crayon
 	if (istype(held_item, /obj/item/toy/crayon))
+		for (var/obj/item/weapon/storage/fancy/crayons/box in user.held_items)
+			for (var/crayon in box)
+				var/obj/item/toy/crayon/c = held_item
```
Comment (jellyveggie2) [inline code/modules/html_interface/paintTool/custom_painting_datum.dm]:
```suggestion
				var/obj/item/toy/crayon/c = crayon
```
This should fix the crayon issue, tested it just now.

**Review by jellyveggie2 — CHANGES_REQUESTED** [review]

**Comment by jellyveggie2** [comment]
@Voltekka, the Closes #35300 should go in the OP text, rather than the title. See #35348 for an example. It should show up in the "Development" section here if done right:

> 


Also, there's no need to credit me in the changelog for a minor review. Keep in mind that changelog is what will go into the in-game changelog later on, verbatim:

>

**Comment by Volttekka** [comment]
Understood

**Comment by Eneocho** [comment]
Thing is this doesn't quite close that feature request. Sure, you get all the colours from a regular crayonbox at once, but you are still limited to the main + shade of the rainbow crayon, which is why deleted the post saying this fixed it.

**Comment by Inorien** [comment]
>clown crayon power creep

i knew it would come one day

**Pull request merged**